generator/test/test_util.dart

```diff
@@ -15,16 +15,18 @@ Map<String, String> getListOfTestFiles(String directory) {
   for (var file in files) {
     if (_path.extension(file.path) == '.dart' && !file.path.endsWith('g.dart')) {
       var expectedFileName = _path.setExtension(file.path, '.expected');
-      if (!files.any((f) => f.path == expectedFileName)) expectedFileName = '';
+       if (!files.any((f) => f.path == expectedFileName)) {
+        throw "Expected file not found. $expectedFileName";
+      }
       result.addAll({_path.basename(file.path): _path.basename(expectedFileName)});
     }
   }
   return result;
 }
 
-Future<dynamic> generatorTestBuilder(String directoryName, String inputFileName, [String expectedFileName = ""]) async {
+Future<dynamic> generatorTestBuilder(String directoryName, String inputFileName, [String? expectedFileName]) async {
```

Comment:
Why is this optional now? If we throw an exception when there's no expected filename, I would imagine this will always be called with some value.